ow_lander/scripts/activity_grind.py

```diff
@@ -7,52 +7,123 @@
+    # Default trenching values
+    x_start = 1.65
+    y_start = 0.0
+    depth = 0.05
+    length = 0.6
+    parallel = True
+    ground_position = constants.DEFAULT_GROUND_HEIGHT
+    delete_prev_traj = False
+
+  else :
+    x_start = req.x
+    y_start = req.y
+    depth = req.depth
+    length = req.length
+    parallel = req.parallel
+    ground_position = req.ground_position
+    delete_prev_traj = req.delete_prev_traj
+
+  return [req.use_defaults, x_start, y_start, depth, length, parallel, ground_position, delete_prev_traj]
+
+
+def plan_cartesian_path(move_group, length, alpha, parallel):
+
+  if parallel==False:
+    alpha = alpha - math.pi/2
 
   waypoints = []
-  wpose = move_arm.get_current_pose().pose
-  wpose.position.x += length
+  wpose = move_group.get_current_pose().pose
+  wpose.position.x += length*math.cos(alpha)
+  wpose.position.y += length*math.sin(alpha)
   waypoints.append(copy.deepcopy(wpose))
 
-  (plan, fraction) = move_arm.compute_cartesian_path(
+  (plan, fraction) = move_group.compute_cartesian_path(
                                waypoints,   # waypoints to follow
                                0.01,        # end effector follow step (meters)
                                0.0)         # jump threshold
+
   return plan, fraction
 
-def grind(move_arm, move_limbs, args):
+def grind(move_arm, move_limbs, move_grinder, args):
 
   x_start = args[1]
-  y_start = args[2]
+  y_start = args[2] 
   depth = args[3]
   length = args[4]
-  ground_position = args[5]
+  parallel = args[5]
+  ground_position = args[6]
 
-  pre_move_complete = move_to_pre_trench_configuration(move_arm, x_start, y_start)
-  if pre_move_complete == False:
-    return False
+  # Compute shoulder yaw angle to trench
+  alpha = math.atan2(y_start-constants.Y_SHOU, x_start-constants.X_SHOU)
+  h = math.sqrt( pow(y_start-constants.Y_SHOU,2) + pow(x_start-constants.X_SHOU,2) )
+  l = constants.Y_SHOU - constants.HAND_Y_OFFSET
+  beta = math.asin (l/h)
+  alpha = alpha+beta
+  
+  if parallel:
+    R = math.sqrt(x_start*x_start+y_start*y_start)
+    # adjust trench to fit scoop circular motion
+    dx = 0.04*R*math.sin(alpha) # Center dig_circular in grind trench 
+    dy = 0.04*R*math.cos(alpha)
+    x_start = 0.9*(x_start + dx) # Move starting point back to avoid scoop-terrain collision
+    y_start = 0.9*(y_start - dy)
+  else:
+    dx = 5*length/8*math.sin(alpha)
+    dy = 5*length/8*math.cos(alpha)
+    x_start = 0.97*(x_start - dx) # Move starting point back to avoid scoop-terrain collision
+    y_start = 0.97*(y_start + dy)   
 
-  # rotate hand
-  change_joint_value(move_arm, constants.J_HAND_YAW, -2*math.pi/3)
+  goal_pose = move_grinder.get_current_pose().pose
+  goal_pose.position.x = x_start
+  goal_pose.position.y = y_start
+  goal_pose.position.z = 0.25 
+  goal_pose.orientation.x = 0.70616885803
+  goal_pose.orientation.y = 0.0303977418722
+  goal_pose.orientation.z = -0.706723318474
+  goal_pose.orientation.w = 0.0307192507001
```

Comment:
Literal numbers used in code are sometimes termed "magic numbers".  It's good to either abstract them with named constants, or add a comment explaining why these values were chosen.

Reply:
You are right! These numbers describe a specific pose (position+orientation) that I want the grinder to have in order to penetrate the terrain vertically and not diagonally as it was before. What do you think is best: comments or putting them in constants.py ?

Reply:
If they are only in this file, comments.  If used by other files, in constants.py.